src/main/java/com/simleetag/homework/api/domain/work/task/api/TaskMaintenanceController.java

```diff
@@ -29,4 +32,30 @@ public ResponseEntity<Long> add(@RequestBody TaskCreateRequest request,
         final TaskGroup taskGroup = taskGroupService.findById(categoryId, taskGroupId);
         return ResponseEntity.ok(taskService.add(taskGroup, request).getId());
     }
+
+    @Operation(summary = "편집")
+    @PutMapping("/{taskId}")
+    public ResponseEntity<TaskResponse> edit(@RequestBody TaskEditRequest request,
+                                             @PathVariable Long categoryId,
+                                             @PathVariable Long taskGroupId,
+                                             @PathVariable Long taskId) {
+        return ResponseEntity.ok(TaskResponse.from(taskService.edit(taskId, request)));
+    }
+
+    @Operation(summary = "단건 조회")
+    @GetMapping("/{taskId}")
+    public ResponseEntity<TaskResponse> searchOne(@PathVariable Long categoryId,
+                                                  @PathVariable Long taskGroupId,
+                                                  @PathVariable Long taskId) {
+        return ResponseEntity.ok(TaskResponse.from(taskService.findById(taskId)));
+    }
+
+    @Operation(summary = "TaskGroup으로 전체 조회")
+    @GetMapping
+    public ResponseEntity<List<TaskResponse>> searchByTaskGroup(@PathVariable Long categoryId,
+                                                     @PathVariable Long taskGroupId) {
+        TaskGroup taskGroup = taskGroupService.findById(taskGroupId);
+        return ResponseEntity.ok(TaskResponse.from(taskService.searchAllByTaskGroup(taskGroup)));
```

Comment:
여기서 taskGroup을 찾지 않고 taskGroupId만을 넘기도록 해도 될 것 같아요.

```suggestion

        return ResponseEntity.ok(TaskResponse.from(taskService.searchAllByTaskGroup(taskGroupId)));
```